chat_client/src/main.py

```diff
@@ -0,0 +1,162 @@
+        self._receiver.start()
+        
+
+    def _start_chat(self) -> None:
+        while True:
+            user = input("\nType user to start chat with or /q to quit: \n").strip()
+            if user == "/q":
+                break
+            if self._receiver.is_unauth():
+                self._close_stream_receiver()
+                break
+            self._start_message_user(user)
+            logging.info("Diconnected from chatroom with user: %s", user)
+        logging.info("Quiting chat app...")
+
+
+    def _start_message_user(self, user: str) -> None:
+        timestamp = Timestamp()
+        logging.info("\nIf you want to quit chatroom, pls type /q")
+        while True:
+            text_to_send = input().strip()
+            if text_to_send == "":
+                continue
+            if text_to_send == "/q":
+                break
+            if self._receiver.is_stopped():
+                logging.warning("Receiver stream closed, trying to reopen...")
+                self._open_stream_receiver()
+            if self._receiver.is_unauth():
+                logging.error("User is not registred")
+                return
+            message = self._create_message(user, text_to_send, timestamp)
+            try:
+                self._stub.SendMessage(request=chat_pb2.SendMessageRequest(message=message))
+            except grpc.RpcError as rpc_error:
+                if rpc_error.code() == grpc.StatusCode.NOT_FOUND:
+                    logging.info("User [%s] not found", user)
+                    self._log_avaible_users()
+                    break
+
+
+    def _create_message(self, user: str, 
+                        text_to_send: str, 
+                        timestamp: Timestamp) -> chat_pb2.Message:      
+        timestamp.GetCurrentTime() 
+        body = chat_pb2.MessageBody(body=text_to_send, timestamp=timestamp.ToJsonString())
+        return chat_pb2.Message(from_user_login=self._login, 
+                                to_user_login=user, 
+                                body=body)          
+                
+
+    def _log_chat_message(self, message: chat_pb2.Message) -> None:
+        logging.info("[%s] %s > %s", 
+                    message.body.timestamp, 
+                    message.from_user_login, 
+                    message.body.body)
+    
+    
+    def disconnect(self) -> None:
+        if not self._is_connected:
+            logging.error("You have to connect first...")
+            return
+        self._close_stream_receiver()
+        self._channel.close()
+        self._channel = None
+        self._stub = None
+        self._is_connected = False
+        logging.info("Disconnected")
+        
+        
+    def _close_stream_receiver(self) -> None:
+        if self._receiver == None:
+            return
+        self._receiver.s_stop()
+        self._receiver.join()
+        self._receiver = None
+
+        
+if __name__ == '__main__':
```

Comment:
Please use double brackets `"` for strings, there is common agreement in Python to use them since Python3 was introduced, and also some static analysis platform consider strings like `'hello, world!'` like a code smell.